pilot/helpers/agents/Developer.py

```diff
@@ -87,36 +106,80 @@ def implement_task(self, i, development_task=None):
+        if self.project.dev_steps_to_load and 'parse_task.prompt' in self.project.dev_steps_to_load[0]['prompt_path']:
+            response = json.loads(self.project.dev_steps_to_load[0]['llm_response']['text'])
+            convo_dev_task.messages = self.project.dev_steps_to_load[0]['messages']
+            remove_last_x_messages = 1  # reason why 1 here is because in db we don't store llm_response in 'messages'
+            self.project.dev_steps_to_load.pop(0)
+        else:
+            response = convo_dev_task.send_message('development/parse_task.prompt', {
+                'running_processes': running_processes,
+                'os': platform.system(),
+                'instructions_prefix': instructions_prefix,
+                'instructions_postfix': instructions_postfix,
+            }, IMPLEMENT_TASK)
+            remove_last_x_messages = 2
+
         steps = response['tasks']
-        convo_dev_task.remove_last_x_messages(2)
+        convo_dev_task.remove_last_x_messages(remove_last_x_messages)
 
         completed_steps = []
 
         while True:
+            # This whole if statement is loading of project.
+            # We want to skip to last iteration that user had in this task (in continue_development function) which is
+            # last iteration.prompt. That prompt must be between current dev step and skip_until_dev_step or last dev
+            # step in db.
+            if self.project.dev_steps_to_load:
+                # get last occurrence of iteration.prompt (used in continue_development)
+                last_iteration = next((el for el in reversed(self.project.dev_steps_to_load) if
+                                       'iteration.prompt' in el.get('prompt_path', '')), None)
+
+                # if no iteration.prompt then finish loading and continue execution normally
+                if last_iteration is None:
+                    self.project.finish_loading()
+                    continue
+
+                # detailed_user_review_goal is explanation for user how to review task (used in continue_development)
+                self.project.last_detailed_user_review_goal = next(
+                    (el for el in reversed(self.project.dev_steps_to_load) if
+                     'define_user_review_goal.prompt' in el.get('prompt_path', '')), None)
+
+                # run_command is command to run app (used in continue_development)
+                if self.run_command is None:
+                    self.run_command = next(
+                        (el for el in reversed(self.project.dev_steps_to_load) if
+                         'get_run_command.prompt' in el.get('prompt_path', '')), None)
+                    if self.run_command is not None:
+                        self.run_command = json.loads(self.run_command['llm_response']['text'])['command']
+
+                self.project.cleanup_list('dev_steps_to_load', last_iteration['id'])
+                self.project.last_iteration = self.project.dev_steps_to_load[0]
```

Comment:
Should we have `self.project.finish_loading()` here? Or are we implicitly trusting that we're in a loop and the next iteration of tle loop won't find the last iteration because we just handled it, and it will call finish loading for us?